src/components/SquareBox/SquareBox.tsx

```diff
@@ -33,8 +35,18 @@ const styles = StyleSheet.create({
 
 export function SquareBox() {
     const id = React.useId();
-        const [squares, setSquares] = React.useState<Array<string>>([]);
-      
+        const [squares, setSquares] = React.useState<Array<Object>>([]);
```

Comment:
You should create a new `interface` (see https://www.typescriptlang.org/docs/handbook/interfaces.html) e.g. `Shape` with properties `color` and `type`, then you can change this line to:

```
const [squares, setSquares] = React.useState<Array<Shape>>([]);
```